inspirehep/records/api/base.py

```diff
@@ -264,8 +264,21 @@ def get_records_by_pids(cls, pids):
         for data in query.yield_per(100):
             yield cls(data.json)
 
+    @classmethod
+    def get_class_for_record(cls, data):
+        type_from_schema = PidStoreBase.get_pid_type_from_schema(data["$schema"])
+        record_class = cls.get_subclasses().get(type_from_schema)
+        if record_class is None:
+            raise WrongRecordSubclass(
```

Comment:
do we really want to have such restriction? Maybe just log?

Reply:
I think this is a good restriction to have as you shouldn't be using a wrong subclass to create a record of a different type. (Eg JobsRecord.create(literature_record) ) . What would be the alternative? To create a LiteratureRecord anyway and just log it?